src/blueapi/service/runner.py

```diff
@@ -152,6 +152,36 @@ def _rpc(
         )
 
 
+def is_valid_return(value: Any, expected_type: type[T]) -> bool:
+    if expected_type is Any:
+        return True
```

Comment:
it's not documented. when would we need this?

why do we use this 'rpc'?

Reply:
We have 2 processes in blueapi: 1 exposes the REST API, and the other holds the RunEngine. The `_rpc` method in this file passes requests from the first to the second. As part of the validation and type hinting, the expected return type of the function is passed, just in case this has changed in the subprocess holding the RunEngine.

If the function that is to be run in the subprocess has a return annotation of `Any`, then whatever the value that is returned by running the function in the subprocess is, it is valid. If the return annotation is `None`, then the returned value is only valid if it is None.
The linked issue was exposed when returning a subscripted list, e.g. `list[DeviceModel]`, as `isinstance(foo, list[DeviceModel])` throws an exception. So this new code is a kind of `isinstance` that recursively finds the origin (`list` in the example) and args (`DeviceModel` in the example) and checks that the returned value is valid for both.

Reply:
@stan-dot https://github.com/DiamondLightSource/blueapi/issues/590

Reply:
can we just have no subprocesses?

https://github.com/DiamondLightSource/blueapi/issues/504

Reply:
This PR makes #504 easier by making the subprocess work more like a separate RPC service...

Reply:
504 isn't difficult in the first place imo, and this feels like reinventing the weel

Reply:
You may be right, but this is the smallest amount of work required to unblock people here and now